core/node/node_framework/src/implementations/layers/metadata_calculator.rs

```diff
@@ -49,15 +68,16 @@ impl WiringLayer for MetadataCalculatorLayer {
             .get_custom(10)
             .await?;
 
-        let object_store = context.get_resource::<ObjectStoreResource>().await.ok(); // OK to be None.
-        if object_store.is_none() {
-            tracing::info!(
-                "Object store is not provided, metadata calculator will run without it."
-            );
-        }
+        let object_store = match self.config.mode {
```

Comment:
will be fair to say it is `object_store_resource`

Reply:
I hope this can be inferred from the context 😅